src/Management/src/Endpoint/Metrics/Observer/EventCounterListener.cs

```diff
@@ -120,27 +120,21 @@ private void ExtractAndRecordMetric(string eventSourceName, IDictionary<string,
         long? longValue = null;
         string counterName = string.Empty;
         var labelSet = new List<KeyValuePair<string, object>>();
-        bool excludedMetric = false;
         string counterDisplayUnit = null;
         string counterDisplayName = null;
 
         foreach (KeyValuePair<string, object> payload in eventPayload)
         {
-            if (excludedMetric)
-            {
-                break;
-            }
-
             string key = payload.Key;
 
             switch (key)
             {
                 case var _ when key.Equals("Name", StringComparison.OrdinalIgnoreCase):
                     counterName = payload.Value.ToString();
 
-                    if (_options.ExcludedMetrics.Contains(counterName))
+                    if ((_options.IncludedMetrics.Any() && !_options.IncludedMetrics.Contains(counterName)) || _options.ExcludedMetrics.Contains(counterName))
```

Comment:
```suggestion
                    // Inclusion list is intended to be evaluated before exclusion list
                    if ((_options.IncludedMetrics.Any()
                        && !_options.IncludedMetrics.Contains(counterName))
                        || _options.ExcludedMetrics.Contains(counterName))
```
Comment doesn't need to be verbatim, but I think we should note the precedence decision